cmd/main.go

```diff
@@ -64,6 +65,12 @@ func Main() {
 		os.Exit(caddy.ExitCodeFailedStartup)
 	}
 
+	undo, err := maxprocs.Set()
+	defer undo()
```

Comment:
What are the consequences if the undo function isn't run? I'm asking because `os.Exit` overrides the `defer` runs and terminates the program immediately. Or maybe we can call `undo` in line 75 (68 before this PR).

Reply:
I think there are no consequences because the program exits.

Reply:
So would undo only be useful in case some weird custom build of Caddy calls `caddycmd.Main()` in a loop or something? I'm not sure how that would even work 🤔

Reply:
I think it's there to allow for momentarily some workloads.

Reply:
That is interesting. But yeah, if the process exits, any side-effects are the responsibility of the OS, so I think this is fine :man_shrugging: